File: Leetcode/191. Number of 1 Bits/191_Number_of_Bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        result = 0
        while n:
            n &= n - 1
            result += 1
        return result


class Main:
    def execute(self):
        solution = Solution()
        n = 11
        print(solution.hammingWeight(n))


main = Main()
main.execute()

File: Leetcode/191. Number of 1 Bits/191_Number_of_Bits.cpp
#include <iostream>
#include <cstdint>
#include <cstdio>
using namespace std;

class Solution
{
public:
    int hammingWeight(uint32_t n)
    {
        int result = 0;
        while (n)
        {
            result += n & 1;
            n >>= 1;
        }
        return result;
    }
};

int main()
{
    Solution s;
    cout << s.hammingWeight(11) << endl;
    return 0;
}